ecscollector/collector.go

```diff
@@ -0,0 +1,270 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package ecscollector implements a Prometheus collector for Amazon ECS
+// metrics available at the ECS metadata server.
+package ecscollector
+
+import (
+	"context"
+	"log"
+
+	"github.com/prometheus-community/ecs_exporter/ecsmetadata"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	numProcsDesc = prometheus.NewDesc(
+		"ecs_num_procs",
+		"Number of processes.",
+		labelKeys, nil)
+
+	cpuTotalDesc = prometheus.NewDesc(
+		"ecs_cpu_total",
+		"Total CPU usage.",
+		labelKeys, nil)
+
+	cpuUserDesc = prometheus.NewDesc(
+		"ecs_cpu_user",
+		"Total CPU usage by user space.",
+		labelKeys, nil)
+
+	cpuKernelDesc = prometheus.NewDesc(
+		"ecs_cpu_kernel",
```

Comment:
For the CPU usage, I'd get closer to the node_exporter metrics ( https://node.demo.do.prometheus.io/metrics )

Maybe something along:

`ecs_task_cpu_seconds_total{cpu="0",mode="idle"}`

Reply:
Done. The metadata server is not providing a breakdown for each CPU, so I'm reporting the total CPU for the container for now. We can introduce a new counter for per CPU metrics.